### ISSUE TITLE
How to computer conrrespondence use the method of "Learning Elementary Structure" 

### ISSUE THREAD

**Issue by cainiaoshidai** [ThibaultGROUEIX/3D-CODED#24]
Dear professor,
Recently, I have been Interested in your papers，and I have some questions. Questions are as follows,
1. I want to compute conrrespondence use templates (Elementary Structure) after trained the AtlasNetV2 ,but I don't know where is the "Elementary Structure". 
2. Now I get two files (network.pth and opt.pickle) ，and I have no idea how to get the "FinalReconstruction.ply".
3. In this paper, your evaluation conrrespondence results use reconstruction loss, but I want to know whether I can use geodetic error to access evaluation conrrespondence results, so I can unify with the functional map method.

**Comment by ThibaultGROUEIX** [comment]
Hi @cainiaoshidai ,

1. You can find the trained models for AtlasNetV2 on FAUST here (same link as in 3D-CODED readme) : 
https://drive.google.com/file/d/1ZAjOuTaeDrKJbFffzLnLn_K-C86fYCXs/view

2. You can just follow the readme to generate results with the trained models.

3. The correspondences are evaluated on an online FAUST server for fairness. There is no access to the ground truth correspondences in the dataset so it would be hard to chance the evaluation metric.

Unifying it with functional maps sounds promising: good luck !
Best,
Thibault

**Comment by cainiaoshidai** [comment]
Thank you!
I just follow the readme to train models(AtlasNetV2). I get two files (network.pth and opt.pickle), but I have no idea how to computer conrrespondence next. Can you give me some advice?

**Comment by ThibaultGROUEIX** [comment]
Thanks, that's a good point! You may want to look at the `Demo` section of the readme. Though I agree it's not clear to look at `demo` for `inference`. I'll update the readme accordingly.

In a nutshell :
* You can use `python inference/correspondences.py --dir_name YOUR_DIRECTORY_RESULTS` to generate result correspondences between two meshes using your trained model
* You can use `python inference/script.py --dir_name YOUR_DIRECTORY_RESULTS --HR 1 --faust INTER` to generate result for all FAUST pairs.

Best regards

**Comment by cainiaoshidai** [comment]
Dear professor,
Thanks for your help. In the paper's code of [Learning Elementary Structure](https://github.com/TheoDEPRELLE/AtlasNetV2), I just get  two files (`network.pth` and `opt.pickle`). I think I can run [inference/correspondences.py](https://github.com/ThibaultGROUEIX/3D-CODED/blob/master/inference/correspondences.py) to compute conrrespondence between source shapes and target shapes use  elementary structure.  But now I meet some questions I will show you in follow.
First, in the paper of [Learning Elementary Structure](https://github.com/TheoDEPRELLE/AtlasNetV2), the elementary structure is obtained through ShapeNet dataset . And the reconstructed shapes are obtained through deformation of the elementary structure. You mentioned use  the elementary structure to instead of the template which mentioned in 3D-CODED. But in your code you did not use elementary structure, so I want to know if I can use elementary structure.

Second, if I can use elementary structure, I want to konw how to use elementary structure to test the `network.pth` use [inference/correspondences.py](https://github.com/ThibaultGROUEIX/3D-CODED/blob/master/inference/correspondences.py).  
Looking forward for your reply. 
Thank you.

**Comment by ThibaultGROUEIX** [comment]
Q1. `But in your code you did not use elementary structure, so I want to know if I can use elementary structure.` : I am not sure I understand what you mean. The training script has two argument flag to use a hand-crafted human template (3D-CODED) or to learn to template as well as the reconstruction (Learning Elementary Structure). In the later case, you can use either `--point_translation 1` or `--patch_deformation 1` or both.

Q2. Once learned, [templates+trained network] are used in the same way to compute correspondences (discussed in past messages).

Best,

**Comment by cainiaoshidai** [comment]
Thanks  professor,
I am glad to get your help! I wish also to thanks for the assistance you have given me in my work!  I think I overlooked some of the things you mentioned. Again, thank you so much.
Best Regards,
